index.js

```diff
@@ -1,39 +1,85 @@
-const dns = require('node:dns');
-const http = require('node:http');
-const os = require('node:os');
-const process = require('node:process');
-const { Buffer } = require('node:buffer');
-const { debuglog } = require('node:util');
-const { getEventListeners, setMaxListeners } = require('node:events');
-const { isIP, isIPv4, isIPv6 } = require('node:net');
-const { toASCII } = require('punycode/');
-const autoBind = require('auto-bind');
-const getStream = require('get-stream');
-const hostile = require('hostile');
-const ipaddr = require('ipaddr.js');
-const isStream = require('is-stream');
-const mergeOptions = require('merge-options');
-const pMap = require('p-map');
-const pWaitFor = require('p-wait-for');
-const packet = require('dns-packet');
-const semver = require('semver');
-const { getService } = require('port-numbers');
-const pkg = require('./package.json');
+import dns from 'node:dns';
+import http from 'node:http';
+import os from 'node:os';
+import process from 'node:process';
+import { Buffer } from 'node:buffer';
+import { debuglog } from 'node:util';
+import { getEventListeners, setMaxListeners } from 'node:events';
+import { isIP, isIPv4, isIPv6 } from 'node:net';
+import { readFileSync } from 'node:fs';
+import { fileURLToPath } from 'node:url';
+import path from 'node:path';
+import punycodeModule from 'punycode/punycode.js';
+import autoBind from 'auto-bind';
+import getStream from 'get-stream';
+import hostile from 'hostile';
+import ipaddr from 'ipaddr.js';
+import { isStream } from 'is-stream';
+import mergeOptions from 'merge-options';
+import pMap from 'p-map';
+import pWaitFor from 'p-wait-for';
+import packet from 'dns-packet';
+import semver from 'semver';
+import portNumbers from 'port-numbers' with { type: 'json' };
+import { request as undiciRequest } from 'undici';
+
+const { toASCII } = punycodeModule;
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
```

Comment:
Why still `__filename` and `__dirname`? If you really need these, it means the package is better stay CommonJS: you are still heavily relying on CommonJS features.